From b57c4407d6438e252ade2b753565da9fffc8f8b7 Mon Sep 17 00:00:00 2001
From: Ahmed Ismail <Ahmed.Ismail@arm.com>
Date: Tue, 2 Jul 2024 10:46:38 +0100
Subject: [PATCH] Do not define __PASTE macro for Arm-compiler

__PASTE macro is defined for Arm Compiler in its header files
so to avoid conflicts it is undefined as it is already defined by
corePkcs11 headers.

Signed-off-by: Ahmed Ismail <Ahmed.Ismail@arm.com>
---
 source/include/core_pkcs11.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/include/core_pkcs11.h b/source/include/core_pkcs11.h
index d8b84f7..6195f3f 100644
--- a/source/include/core_pkcs11.h
+++ b/source/include/core_pkcs11.h
@@ -197,8 +197,8 @@
 
 /* Bring in the public header. */
 
-/* Undefine the macro for Keil Compiler to avoid conflict */
-#if defined( __PASTE ) && defined( __CC_ARM )
+/* Undefine the macro for Keil and ARMClang Compilers to avoid conflict */
+#if defined( __PASTE ) && ( defined( __CC_ARM ) || defined( __ARMCC_VERSION ) )
     /* ARM RCVT stdint.h has a duplicate definition with PKCS #11. */
     #undef __PASTE
 #endif
-- 
2.34.1

